src/libcore/time.rs

```diff
@@ -263,6 +263,57 @@ impl Duration {
     #[inline]
     pub fn subsec_nanos(&self) -> u32 { self.nanos }
 
+    /// Returns the total number of milliseconds contained by this `Duration`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #![feature(duration_as_u128)]
+    /// use std::time::Duration;
+    ///
+    /// let duration = Duration::new(5, 730023852);
+    /// assert_eq!(duration.as_millis(), 5730);
+    /// ```
+    #[unstable(feature = "duration_as_u128", issue = "50202")]
+    #[inline]
+    pub fn as_millis(&self) -> u128 {
+        self.secs as u128 * MILLIS_PER_SEC as u128 + self.nanos as u128 / NANOS_PER_MILLI as u128
```

Comment:
Nit: `self.nanos as u128 / NANOS_PER_MILLI as u128` should be replaced with `(self.nanos / NANOS_PER_MILLI) as u128`. The former requires doing 128-bit arithmetic on a 32-bit number, which is redundant.